agent/src/agent/streamsets/streamsets.py

```diff
@@ -0,0 +1,261 @@
+import json
+import requests
+import time
+import urllib.parse
+
+from sqlalchemy import Column, Integer, String
+from agent.modules.db import Entity
+from agent.modules.logger import get_logger
+from agent.modules.constants import STREAMSETS_PREVIEW_TIMEOUT
+
+logger = get_logger(__name__)
+
+
+def endpoint(func):
+    """
+    Decorator for api endpoints. Logs errors and returns json response
+
+    :param func:
+    :return:
+    """
+
+    def wrapper(*args, **kwargs):
+        res = func(*args, **kwargs)
+        try:
+            res.raise_for_status()
+            if res.text:
+                return res.json()
+            return
+        except requests.exceptions.HTTPError:
+            if res.text:
+                try:
+                    response = res.json()
+                    logger.exception(response['RemoteException'])
+                    if res.status_code == 401:
+                        raise UnauthorizedException('Unauthorized')
+                    else:
+                        raise ApiClientException(
+                            response['RemoteException']['message'],
+                            response['RemoteException']['exception'],
+                        )
+                except json.decoder.JSONDecodeError:
+                    raise ApiClientException(res.text)
+            raise
+
+    return wrapper
+
+
+class StreamSets(Entity):
+    __tablename__ = 'streamsets'
+
+    id = Column(Integer, primary_key=True)
+    url = Column(String)
+    username = Column(String)
+    password = Column(String)
+
+    def __init__(self, url, username, password):
+        self.url = url
+        self.username = username
+        self.password = password
+
+
+class StreamSetsApiClient:
```

Comment:
so why did you move it from a separate file?

Reply:
I didn't move it from a separate file, I moved that file entirely from modules directory, renamed it to streamsets and added StreamSets entity